MediaAdminBundle/Controller/Api/MediaController.php

```diff
@@ -90,4 +91,71 @@ public function deleteAction($mediaId)
 
         return array();
     }
+
+    /**
+     * @param Request $request
+     * @param string  $folderId
+     * 
+     * @Config\Route("/upload/{folderId}", name="open_orchestra_api_media_upload")
+     * Config\Method({"POST"})
+     * 
+     * @return FacadeInterface
+     */
+    public function uploadAction($folderId, Request $request)
+    {
+        $uploadedFile = $request->files->get('file');
+
+        if ($uploadedFile) {
+            $tmpDir = $this->getParameter('open_orchestra_media.tmp_dir');
```

Comment:
I would suggest to extract the logic part from the controller and put it inside a service